console/test.ts

```diff
@@ -21,91 +24,136 @@ function restorePrint(cb: () => void): void {
+            }
+        }
+        console.log(`Passed ${numPassed}/${numTests} tests.`); 
+    });
+}
+
+// Add more tests here:
+var testList = [
+    [`\n--- Math tests ---`],
+    ["Unary operations test", "pytests/math/unaryOpsTest"],
+    ["Binary operations test", "pytests/math/binaryOpsTest"],
+    ["In-place operations test", "pytests/math/inplaceTest"],
+    ["Mixed Arithmetic test", "pytests/math/mixedMathTest"],
+    [`\n--- Function tests ---`],
+    ["Keyword and default arguments test", "pytests/functions/keywordargs"],
+    ["Recursion test", "pytests/functions/recursionTest"],
+    ["Scoping test", "pytests/functions/scopeTest"],
+    ["Generators test", "pytests/functions/generatorTest"],
+    [`\n--- Builtin tests ---`],
+    ["Builtin Types test", "pytests/builtins/builtinTypes"],
+    ["Bin function test", "pytests/builtins/bin"],
+    ["Hex function test", "pytests/builtins/hex"],
+    ["Complex function test", "pytests/builtins/complex"],
+    ["Bool function test", "pytests/builtins/bool"],
+    ["Divmod function test", "pytests/builtins/divmod"],
+    ["Abs function test", "pytests/builtins/abs"],
+    ["Cmp function test", "pytests/builtins/cmp"],
+    ["All & Any functions test", "pytests/builtins/all_any"],
+    ["Chr & Ord functions test", "pytests/builtins/chr_ord"],
+    ["Attribute accessors test", "pytests/builtins/attrs"],
+    ["Underscore names test", "pytests/builtins/underscoresTest"],
+    [`\n--- Collection tests ---`],
+    ["List test", "pytests/collections/lists"],
+    ["Set test", "pytests/collections/sets"],
+    ["Dict test", "pytests/collections/dicts"],
+    [`\n--- Control flow tests ---`],
+    ["Loop test", "pytests/loopTest"],
+    ["Range test", "pytests/rangeTest"],
+    ["Comprehension test", "pytests/comprehensionTest"],
+    [`\n--- Class tests ---`],
+    ["Basic class test", "pytests/classes/userDefTest"],
+    [`\n--- Caught Exception tests ---`],
+    ["Basic Exception test", "pytests/caught_exceptions/except"],
+    ["Nested Function Exception test", "pytests/caught_exceptions/nestedFunction"],
+    ["Nested Try-Block Exception test", "pytests/caught_exceptions/nestedTry"],
+    ["Recursive function Try-Block Exception test", "pytests/caught_exceptions/recursiveFunctionTry"],
+    [`\n--- Uncaught Exception tests ---`],
+    ["Basic Uncaught Exception test", "pytests/uncaught_exceptions/simpleFunction"],
+    ["Recurisve Function Uncaught Exception test", "pytests/uncaught_exceptions/recursiveFunctionUncaught"],
+    [`\n--- Thread tests ---`],
+    ["Termination test", "pytests/threads/terminationTest"],
+    [`\n--- Other tests ---`],
+    ["Strings test", "pytests/stringTest"],
+    ["Slice test", "pytests/sliceTest"],
+    ["Assignment test", "pytests/assignmentTest"],
+    ["Deletion test", "pytests/delTest"]
+];
+var testList_dup = testList.slice(0);
+
+var d = domain.create();
+d.on('error', function(err){
+    onFailure();
```

Comment:
Whenever an exception occurs, ```d.on('error', function(err))``` catches the error. But for some reason, the ```async.eachSeries``` execution halts. I have tried it various ways, but the ```async.eachSeries``` terminates prematurely, whenever an exception is raised. But if you call the ```processTests``` in the domain error handler, you can resume the execution of the remaining tests.

I achieve it through the following way.

I duplicate the ```testList``` into ```testList_dup``` before any tests are run.
In the ```iteration``` function, the tests which are being processed are immediately removed from ```testList_dup```. So if an error occurs, and the domain halts the execution, we know where to resume our tests from. The ```testList``` is then assigned the tests which have not run yet, inside the ```d.on('error,..)```. Now we run ```processTests``` and it runs the remaining tests.

So every test is run only once.